core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -827,9 +868,21 @@ class DAGScheduler(
     // might modify state of objects referenced in their closures. This is necessary in Hadoop
     // where the JobConf/Configuration object is not thread-safe.
     var taskBinary: Broadcast[Array[Byte]] = null
+
+    // Check if RDD serialization debugging is enabled
+    val debugSerialization: Boolean = sc.getConf.getBoolean("spark.serializer.debug", false)
```

Comment:
Rather than having a config option for this, why not just always run this debugging output _after_ we've seen a serialization failure? The performance overhead won't matter much if we do it after a failure only.